crypto/src/vrf/mod.rs

```diff
@@ -151,6 +162,80 @@ impl VRFPublicKey {
     }
 }
 
+/// Given a tree of keys that are derived from a master key using BIP32 rules, this struct represents
+/// the private key at one of the nodes of this tree.
+#[derive(Debug, PartialEq, Eq, Clone, Encode, Decode)]
+pub struct ExtendedVRFPrivateKey {
+    /// The absolute derivation path that was used to derive this key
+    derivation_path: DerivationPath,
+    /// The chain code is used in BIP32 in conjunction with the private key to allow derivation
+    /// of child keys
+    chain_code: ChainCode,
+    /// The private key to be used to derive child keys of this node using BIP32 rules
+    private_key: VRFPrivateKey,
+}
+
+impl ExtendedVRFPrivateKey {
+    pub fn new_master(
+        seed: &[u8],
+        keykind: VRFKeyKind,
+    ) -> Result<ExtendedVRFPrivateKey, DerivationError> {
+        // Create a new mac with the appropriate BIP39 constant
+        let mut mac = new_hmac_sha_512(b"Bitcoin seed");
```

Comment:
Maybe it would be better to separate the derivation starting from the root (using different arguments to `new_hmac_sha_512`)?
That way we can be a bit more confident that the keys will not interfere with each other. 

```
impl Secp256k1ExtendedPrivateKey {
    pub fn new_master(seed: &[u8]) -> Result<Secp256k1ExtendedPrivateKey, DerivationError> {
        // Create a new mac with the appropriate BIP39 constant
        let mut mac = new_hmac_sha_512(b"Bitcoin seed");

impl ExtendedVRFPrivateKey {
    pub fn new_master(
        seed: &[u8],
        keykind: VRFKeyKind,
    ) -> Result<ExtendedVRFPrivateKey, DerivationError> {
        // Create a new mac with the appropriate BIP39 constant
        let mut mac = new_hmac_sha_512(b"Bitcoin seed");
```

Probably no big deal, since VRFs use a different path at any case.

Reply:
Nice suggestion, this makes sense to me as with this even if the Secp256k1 master key is leaked the VRF master key and children will still be safe.
@TheQuantumPhysicist @muursh

Reply:
I don't mind, sounds like a good idea. Though keep in mind that the secp256k1 derivation shouldn't change because it's compatible with hardware wallets, and I'm not sure how hard it'll be to change that. 

Let @muursh also provide his opinion.

Reply:
Yeah I don't see an issue with that

Reply:
secp256k1 remains the same as before, I only changed the constant for the VRF master key derivation to "Bitcoin VRF seed"